pkg/kotsutil/kots.go

```diff
@@ -396,6 +405,74 @@ func (o KotsKinds) Marshal(g string, v string, k string) (string, error) {
 	return "", errors.Errorf("unknown gvk %s/%s, Kind=%s", g, v, k)
 }
 
+func (k *KotsKinds) addKotsKind(content []byte) error {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+
+	// kots kinds could be part of a multi-yaml doc
+	docs := util.ConvertToSingleDocs(content)
+	for _, doc := range docs {
+		decoded, gvk, err := decode(doc, nil, nil)
+		if err != nil {
+			return errors.Wrap(err, "failed to decode yaml content")
+		}
+
+		if strings.HasPrefix(gvk.String(), "troubleshoot.replicated.com/v1beta1,") {
+			doc, err = docrewrite.ConvertToV1Beta2(doc)
+			if err != nil {
+				return errors.Wrap(err, "failed to convert to v1beta2")
+			}
+			decoded, gvk, err = decode(doc, nil, nil)
+			if err != nil {
+				return err
+			}
+		}
+
+		switch gvk.String() {
+		case "kots.io/v1beta1, Kind=Config":
+			k.Config = decoded.(*kotsv1beta1.Config)
+		case "kots.io/v1beta1, Kind=ConfigValues":
+			k.ConfigValues = decoded.(*kotsv1beta1.ConfigValues)
+		case "kots.io/v1beta1, Kind=Application":
+			k.KotsApplication = *decoded.(*kotsv1beta1.Application)
+		case "kots.io/v1beta1, Kind=License":
+			k.License = decoded.(*kotsv1beta1.License)
+		case "kots.io/v1beta1, Kind=Identity":
+			k.Identity = decoded.(*kotsv1beta1.Identity)
+		case "kots.io/v1beta1, Kind=IdentityConfig":
+			k.IdentityConfig = decoded.(*kotsv1beta1.IdentityConfig)
+		case "kots.io/v1beta1, Kind=Installation":
+			k.Installation = *decoded.(*kotsv1beta1.Installation)
+		case "kots.io/v1beta1, Kind=HelmChart":
+			if k.HelmCharts == nil {
+				k.HelmCharts = &kotsv1beta1.HelmChartList{}
```

Comment:
why not just use the same type defined in the KotsKinds struct:
```suggestion
				k.HelmCharts = []*kotsv1beta1.HelmChart{}
```
that way you don't have to worry about using `.Items`

Reply:
you actually don't even have to check if it's nil. the previous code should just work, no?
```
kotsKinds.HelmCharts = append(kotsKinds.HelmCharts, decoded.(*kotsv1beta1.HelmChart))
```

Reply:
Regretfully I think there's no other way to add generated k8s apimachinery types and methods to the `[]*kotsv1beta1.HelmChart{}` type.